### PR TITLE
Added quick peek

### PR DESCRIPTION
**Pull request by SK0lll**
There is still room for improvement but a basic quick peek should help with some HvH.

Small things I want to add if I have the time:

- [x] Better visuals
- [x] Fix anything broken (Nothing right now but that can always change)

### CODE REVIEW

File: NEPS/Config.cpp [hunk 1]
```diff
@@ -1292,6 +1294,8 @@ static void to_json(json &j, const Config::Movement &o)
 	WRITE("Auto strafe", autoStrafe);
 	WRITE("Edge jump", edgeJump);
 	WRITE("Fast stop", fastStop);
+	WRITE("QuickPeekKey", QuickPeekKey);
```
Comment (degeneratehyperbola) [inline NEPS/Config.cpp:1297]: Here it is "QuickPeekKey" but when reading it is the same but with spaces "Quick Peek Key", this causes it to fail to load and reset to its default value

File: NEPS/Hacks/Misc.cpp [hunk 2]
```diff
@@ -1859,3 +1870,75 @@ void Misc::fakePrime() noexcept
...
+			//cmd->viewangles = usercmds_for_quickpeek.at(qp_count).viewAngles;
+			//g_EngineClient->SetViewAngles(&cmd->viewangles);
+			qp_count--;
+		}
+	}
+	else
+	{
+		qp_count = usercmds_for_quickpeek.size();
+	}
+}
```
Comment (degeneratehyperbola) [inline NEPS/Hacks/Misc.cpp:1910]: Oh, I see you did the way I would, even though it's a bit more complex, you managed to do it right, my respect 👍🏿

File: NEPS/Hacks/Misc.cpp [hunk 3]
```diff
@@ -38,6 +38,17 @@
...
+struct custom_cmd
+{
+	float			viewangles;
+	float			forwardmove;
+	float			sidemove;
+	float			upmove;
+};
+bool hasShot;
+int qp_count;
+std::vector<custom_cmd>usercmds_for_quickpeek;
```
Comment (degeneratehyperbola) [inline NEPS/Hacks/Misc.cpp:50]: There is no practical reason to do so but just to keep the overall coding style, you should make these vars static

**Review by degeneratehyperbola — CHANGES_REQUESTED** [review]
Amazing, didn't try it out yet, hope to see it working 👍🏿

**Comment by wuuxiehacker303** [comment]
I remember the third task to be completed is“"Mode", &config->movement.quickpeek.mode, "Direct\0Backstep\0"”，Why did cancel the update；     0v0

**Comment by degeneratehyperbola** [comment]
Ok I am merging it and am going to polish it up a bit afterwards. You don't seem to follow the Osiris's code style but that's just nitpicking at this point. Good work!

**Pull request merged**